subsys/net/l2/openthread/openthread_utils.c

```diff
@@ -105,10 +105,17 @@ void add_ipv6_addr_to_zephyr(struct openthread_context *context)
 				       buf, sizeof(buf))));
 		}
 
-		if_addr = net_if_ipv6_addr_add(
+		if (is_mesh_local(context, address->mAddress.mFields.m8)) {
```

Comment:
I'm afraid we need a more sophisticated solution here. The main blocker for me is that now we end up with a link-local address registered as DHCP address:
```
IPv6 unicast addresses (max 6):
        fe80::fcab:bf4:bfe2:693b DHCP preferred infinite
        fdde:ad00:beef:0:80a0:e2b0:cdf3:ccdb autoconf preferred infinite meshlocal
        fdde:ad00:beef::2 manual preferred infinite meshlocal
```
Other thing is that global addresses are not necessarily DHCP assigned - they can be autoconfigured with SLAAC. The  DHCP label does not fit here either.

I've briefly checked OT APIs, and unfortunately, I have not found anything useful that could help to determine the address origin. Perhaps `otNetifAddress` should carry such information, just as Zephyr does?

Reply:
@rlubos  Understand the issue with it. Having link-local assumed to be a DHCP address did not have an impact for us though.

Could also add an additional check for is_link_local() for the moment.

Yes, agree I would add this in upstream to otNetifAddress as well to solve it correctly.

Reply:
> Having link-local assumed to be a DHCP address did not have an impact for us though.

Having an impact or not, it's just incorrect and confusing.

> Could also add an additional check for is_link_local() for the moment.

This should be a reasonable solution for now.

Reply:
Created https://github.com/openthread/openthread/pull/5019, let's see whether that is acceptable for upstream openthread.

Reply:
@rlubos https://github.com/openthread/openthread/pull/5019 has been merged. Is there an zephyr-openthread update planned already? If so, I would wait with an update of this PR.

Reply:
@markus-becker-tridonic-com Thanks for addressing this in OpenThread! I'll try to prepare an upmerge of OpenThread repo tomorrow so we can move this forward.

Reply:
@markus-becker-tridonic-com The OpenThread update was just merged into Zephyr.

Reply:
@rlubos I have pushed an update of this. It compiles, but I still need to test the updated change on a device.

Reply:
output from `net ipv6` looks good:
```
[snipped]
IPv6 addresses for interface 0x20000680 (OpenThread)
====================================================
Type            State           Lifetime (sec)  Address
autoconf        preferred       infinite        fe80::94ef:67c8:e205:794f/128
autoconf        preferred       infinite        fdde:ad00:beef:0:b0e8:5070:edb4:67f/128
manual          preferred       infinite        fdde:ad00:beef::1/128
```
do not have a DHCPv6-enabled BR here right now to check the DHCP path.

Reply:
Yep, looks fine, thanks!